ext/json/ext/generator/generator.c

```diff
@@ -189,63 +242,142 @@ static inline void convert_UTF8_to_JSON(FBuffer *out_buffer, VALUE str, const un
 
     unsigned long beg = 0, pos = 0;
 
-#define FLUSH_POS(bytes) if (pos > beg) { fbuffer_append(out_buffer, &ptr[beg], pos - beg); } pos += bytes; beg = pos;
+#ifdef HAVE_ARM_NEON_H
+    const uint8x16_t lower_bound     = vdupq_n_u8(32); 
+    const uint8x16_t backslash       = vdupq_n_u8(92);
+    const uint8x16_t dblquote        = vdupq_n_u8(34);
+
+    while (pos+16 < len) {
+        uint8x16_t chunk    = vld1q_u8((const uint8_t*)&ptr[pos]);
+        uint8x16_t too_low  = vcltq_u8(chunk, lower_bound);
+
+        uint8x16_t has_backslash = vceqq_u8(chunk, backslash);
+        uint8x16_t has_dblquote = vceqq_u8(chunk, dblquote);
+
+        uint8x16_t invalid = too_low;
+        uint8x16_t has_escaped_char = vorrq_u8(has_backslash, has_dblquote);
+        
+        invalid = vorrq_u8(invalid, has_escaped_char);
+
+        if (vmaxvq_u8(invalid) == 0) {
+            pos += 16;
+            continue;
+        }
+
+        uint8x16_t tmp = vandq_u8(too_low, vdupq_n_u8(0x1));
+        tmp = vorrq_u8(tmp, vandq_u8(has_backslash, vdupq_n_u8(0x2)));
+        tmp = vorrq_u8(tmp, vandq_u8(has_dblquote, vdupq_n_u8(0x4)));
+
+        uint8_t arr[16];
+        vst1q_u8(arr, tmp);
```

Comment:
AH actually, I suppose that's what i was thinking about. It tells you the position of the backslashes and quotes.